src/cargo/core/resolver/features.rs

```diff
@@ -0,0 +1,534 @@
+//! Feature resolver.
+//!
+//! This is a new feature resolver that runs independently of the main
+//! dependency resolver. It is intended to make it easier to experiment with
+//! new behaviors. When `-Zfeatures` is not used, it will fall back to using
```

Comment:
This would be a good place to explain that it is not going to replace the dependency resolver but that it acts as a second pass. By not considering changing which versions are selected it avoids solving the NP-Hard problem faced by the dependency resolver. And it is focused on what should be built for this system not `all possible systems` that the dependency resolver considers. It therefore has complexity budget to consider ... (the list of really cool things your system achieves.)

Reply:
Posted #7918 if you could review at your leisure.

> And it is focused on what should be built for this system not `all possible systems` that the dependency resolver considers.

This part I didn't quite follow.  The dependency resolver runs twice, once with all features enabled (which I think is what you mean by "all possible systems"?), and again with just the features on the command-line.  The new feature resolver has the same capabilities, but it only runs once.  By "same", I mean if you do `--all-features --workspace`, then that should result in the same "all possible" scenario matching `Cargo.lock`'s output.

Reply:
I think I meant a case like:
```
a depends on b and c
b depends on e with feature eb
c depends on e only on MacOS e with feature ec 
e exists
```
What features does `e` have when built on a Windows system? `dependency resolver` thinks `{eb, ec}`. I think, the new features resolver can see that no active dependency asked for `ec`. Is that true?

Reply:
Correct, `ec` is not activated.